### PR TITLE
Add minimal version of new `contour_labels` filter

### PR DESCRIPTION
**Pull request by user27182**
### Overview

Split from #6732. This PR adds a minimal version of that filter to make it easier to review. Many keywords and tests have been trimmed down or removed. The intention to add the other features in follow-up PRs.

### CODE REVIEW

**Comment by tkoyama010** [comment]
Memo: [Document link to review](https://67660d6443e3a06da74822e3--pyvista-dev.netlify.app/api/core/_autosummary/pyvista.imagedatafilters.contour_labels#pyvista.ImageDataFilters.contour_labels)

File: tests/core/test_imagedata_filters.py [hunk 1]
```diff
@@ -166,6 +167,180 @@ def test_contour_labeled_with_invalid_scalars():
         label_map.contour_labeled()
 
 
+@pytest.fixture
+def channels():
+    # ImageData will cell data
+    return examples.load_channels()
```
Comment (tkoyama010) [inline tests/core/test_imagedata_filters.py:173]: Typo?
```suggestion
    # ImageData with cell data
    return examples.load_channels()
```

File: tests/core/test_imagedata_filters.py [hunk 2]
```diff
@@ -166,6 +167,180 @@ def test_contour_labeled_with_invalid_scalars():
...
+    'select_inputs',
+    [None, 2, 5, [2, 5]],
+    ids=['in_None', 'in_2', 'in_5', 'in_2_5'],
+)
+@pytest.mark.parametrize(
+    'select_outputs',
+    [None, 2, 5, [2, 5]],
+    ids=['out_None', 'out_2', 'out_5', 'out_2_5'],
+)
+@pytest.mark.parametrize('boundary_style', ['all', 'external'])  # , 'internal'],
```
Comment (tkoyama010) [inline tests/core/test_imagedata_filters.py:271]: Please add a comment as to why we are skipping `internal` .

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

File: tests/core/test_imagedata_filters.py (hunk 2 above)
Comment (user27182) [reply tests/core/test_imagedata_filters.py]: This omission is probably an oversight, and the tests probably need an additional branch to accommodate this case (e.g. set the expected output). 

But, this case _is_ covered elsewhere with plotting tests (not included in this PR)...

Comment (user27182) [reply tests/core/test_imagedata_filters.py]: Should be fixed with [f5b2adb](https://github.com/pyvista/pyvista/pull/7000/commits/f5b2adb8b087e25fc5cb5cca11a4f4e2d6c13380)

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Pull request merged**